addons/l10n_ke_edi_oscu_mrp/models/mrp_bom.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.exceptions import UserError
+
+from odoo import api, fields, models, _, Command
+
+import json
+
+class MrpBom(models.Model):
+    """ Defines bills of material for a product or a product template """
+    _inherit = 'mrp.bom'
+
+    fiscal_country_codes = fields.Char(compute='_compute_fiscal_country_codes')
+    l10n_ke_validation_message = fields.Json(compute='_compute_l10n_ke_validation_message')
+
+    @api.depends('company_id')
+    @api.depends_context('allowed_company_ids')
+    def _compute_fiscal_country_codes(self):
+        for record in self:
+            allowed_companies = record.company_id or self.env.companies
+            record.fiscal_country_codes = ",".join(allowed_companies.mapped('account_fiscal_country_id.code'))
+
+    # TODO - this is horrible, and might also be a performance nightmare. Though there might be no other way to do this considering the exact behaviour we want
+    @api.depends(
+        'product_id.l10n_ke_packaging_unit_id',
+        'product_id.l10n_ke_packaging_quantity',
+        'product_id.l10n_ke_origin_country_id',
+        'product_id.l10n_ke_tax_type_code',
+        'product_id.l10n_ke_product_type_code',
+        'product_id.uom_id.l10n_ke_quantity_unit_id.code',
+        'product_tmpl_id.product_variant_ids.l10n_ke_packaging_unit_id',
+        'product_tmpl_id.product_variant_ids.l10n_ke_packaging_quantity',
+        'product_tmpl_id.product_variant_ids.l10n_ke_origin_country_id',
+        'product_tmpl_id.product_variant_ids.l10n_ke_tax_type_code',
+        'product_tmpl_id.product_variant_ids.l10n_ke_product_type_code',
+        'product_tmpl_id.product_variant_ids.uom_id.l10n_ke_quantity_unit_id.code',
+        'bom_line_ids.product_id',
+        'bom_line_ids.product_id.l10n_ke_packaging_unit_id',
+        'bom_line_ids.product_id.l10n_ke_packaging_quantity',
+        'bom_line_ids.product_id.l10n_ke_origin_country_id',
+        'bom_line_ids.product_id.l10n_ke_tax_type_code',
+        'bom_line_ids.product_id.l10n_ke_product_type_code',
+        'bom_line_ids.product_id.uom_id.l10n_ke_quantity_unit_id.code',
+    )
```

Comment:
as far as I can see, the dependencies are needed, but the performance impact should not be very big (since the field is not stored, any change in one of the dependencies will just invalidate the cache)